### PR TITLE
Fetch headers

### LINKED ISSUE #478: Ability to set Headers on asset requests
**Is your feature request related to a problem? Please describe.**
I am generating a PDF which would include images that require login and headers set to access.

**Describe the solution you'd like**
The ability to give the renderer a `AssetHeaders` (or similar) object. This way it will be possible to set the `Authorization` header. Then pass it along as headers to `fetch` here: https://github.com/diegomura/react-pdf/blob/master/src/utils/image.js#L159

**Describe alternatives you've considered**
Right now I'm contemplating if I have to download all the files locally, and then fetch them from the local filesystem. But this will not work for the client-side render.

**Additional context**
Nothing of note.

### PR DESCRIPTION
**Pull request by diegomura**
Fixes  #478

### CODE REVIEW

File: tests/image.test.js [hunk 1]
```diff
@@ -77,6 +77,30 @@ describe('Image', () => {
     expect(dummyRoot.instance.image.mock.calls[0][0]).toBe(image.image.data);
   });
 
+  test('Should render a remote image from src object', async () => {
```
[the remark this answers is not in the document]
Comment (diegomura) [reply tests/image.test.js]: Thanks! Agreed. Better to make that check in the `Image` component 👍

**Comment by diegomura** [comment]
Thanks for the review @DuncanMacWeb !

I absolutely agree that lining up with RN api is a great idea, but I have some doubts about supporting both `src` and `source` as valid props to define the image actual source. Would be basically an `alias` right? I'm not sure about having two props that basically mean the same thing, even though I understand that adding it will respect RN API completely.

Thoughts? Is there any other reason you think we should support `source` also?

**Comment by DuncanMacWeb** [comment]
Aligning with RN was my only reason to keep `src` and `source` separate, yes… I do think it’s valuable to have web and RN compatibility, particularly for RN, as your interface already largely follows its conventions. I agree it would create some duplication, but I expect people who don’t study the docs (or just want to render some existing code to PDF quickly) will tend to use `Image` the way they’re used to: which is, either `src` if they’re used to web or `source` if they’re used to RN (or reusing RN code). But happy to leave the decision with you :)

**Comment by diegomura** [comment]
Yep. I think you are right about this. It doesn't harm to support both, and I have to admit that it's very cool when a lib works based on things or props that you _guess_ based on your previous experience with a similar API. 

Trying to wrap this PR today, so then I can focus on the Firefox issue you created!

**Comment by DuncanMacWeb** [comment]
Amazing work, thank you @diegomura, much appreciated! :)

**Comment by diegomura** [comment]
@DuncanMacWeb I think it's done!

I had to get rid of the `cross-fetch -> fetch` alias because I couldn't make it work with `jest-fetch-mock`. Even though I liked that alias, I don't think we are going to switch to another fetch lib soon, and at least this way is clearer about what we are using.

Having fetch mocked is really helpful because it enables to test things such as headers, methods or body (both for `Image` and `Font`) and at the same time it makes the tests run faster and more securely since we are not actually fetching assets over the web.

Would appreciate a review if you don't mind!

File: src/elements/Image.js [hunk 2]
```diff
@@ -7,9 +7,6 @@ import { resolveObjectFit } from '../utils/objectFit';
 
 const SAFETY_HEIGHT = 10;
 
-// We manage two bounding boxes in this class:
-//  - Yoga node: Image bounding box. Adjust based on image and page size
-//  - Image node: Real image container. In most cases equals Yoga node, except if image is bigger than page
```
[the remark this answers is not in the document]
Comment (diegomura) [reply src/elements/Image.js]: They are outdated 😅. With the addition of the `object-fit` support this is not accurate anymore.

File: index.d.ts [hunk 3]
```diff
@@ -87,9 +89,17 @@ declare module '@react-pdf/renderer' {
      */
     class View extends React.Component<ViewProps> {}
 
+    type HTTPMethod = 'GET' | 'HEAD' | 'POST' | 'PUT' | 'DELETE' | 'PATCH';
+
+    type SourceObject =
```
[the remark this answers is not in the document]
Comment (diegomura) [reply index.d.ts]: Thanks! This is much appreciated!
I know nothing about typescript (maybe time to learn? I've been avoiding it for awhile), so I did what I could, but would be great if @yenbekbay or @ishan993 can review this 😄

[the remark this answers is not in the document]
Comment (diegomura) [reply index.d.ts]: This is not entire correct. Maybe I didn't mentioned it before, but `source` is just an alias of `src`. You can pass a `{ uri: string, method: HTTPMethod, headers: {} }` both to `src` or `source` I realized it was better this way, codewise and from the user point of view also.

**Pull request merged**